### PR TITLE
[WIP] Quota test via service catalog

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
Added new test for quota via service dialogs
{{pytest: -vvvv cfme/tests/infrastructure/test_infra_quota_via_services.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 2d61194e4d4c7370c098dafae2110276ed816d21:

`cfme/tests/infrastructure/test_infra_quota_via_services.py`:
- :bangbang: Line 34:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 58:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 60:11: **E221** *multiple spaces before operator*
- :bangbang: Line 63:37: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 68:37: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 71:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 76:37: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 81:37: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 84:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 97:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 111:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 128:15: **E231** *missing whitespace after ';'*
- :bangbang: Line 128:15: **E702** *multiple statements on one line (semicolon)*
- :bangbang: Line 132:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 132:101: **E501** *line too long (103 > 100 characters)*
- :bangbang: Line 133:101: **E501** *line too long (104 > 100 characters)*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3205d5a2e8cb23d14151920cd74a08327c35ec7a:

`cfme/tests/infrastructure/test_infra_quota_via_services.py`:
- :bangbang: Line 36:58: **E226** *missing whitespace around arithmetic operator*
- :bangbang: Line 37:65: **E226** *missing whitespace around arithmetic operator*
- :bangbang: Line 70:15: **E702** *multiple statements on one line (semicolon)*
- :bangbang: Line 71:27: **E261** *at least two spaces before inline comment*
- :bangbang: Line 71:28: **E262** *inline comment should start with '# '*
- :bangbang: Line 75:33: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 80:33: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 85:15: **E261** *at least two spaces before inline comment*
- :bangbang: Line 85:16: **E262** *inline comment should start with '# '*
- :bangbang: Line 89:33: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 94:33: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 123:1: **E265** *block comment should start with '# '*
- :bangbang: Line 153:15: **E702** *multiple statements on one line (semicolon)*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5e191b83d87f5d1ac5799527a4e1361f4232eff0:

`cfme/tests/infrastructure/test_infra_quota_via_services.py`:
- :bangbang: Line 73:47: **E127** *continuation line over-indented for visual indent*
- :bangbang: Line 78:47: **E127** *continuation line over-indented for visual indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by vatsalparekh** [comment]
@RonnyPfannschmidt need your review about the use of `parametrize` with `indirect` in this

**Comment by dajoRH** [comment]
Lint report for commit a21d76c9600cf58ec543eda3564e3f9bce55a0c8:

`cfme/tests/infrastructure/test_infra_quota_via_services.py`:
- :bangbang: Line 70:43: **E201** *whitespace after '{'*
- :bangbang: Line 70:67: **E202** *whitespace before '}'*
- :bangbang: Line 73:43: **E201** *whitespace after '{'*
- :bangbang: Line 73:64: **E202** *whitespace before '}'*
- :bangbang: Line 118:1: **E302** *expected 2 blank lines, found 1*

Hmmm
:frowning: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by vatsalparekh** [comment]
This is supposed to skips tests, because of the BZ not fixed yet.

File: cfme/tests/infrastructure/test_infra_quota_via_services.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+@pytest.fixture(scope="function")
+def template_name(provisioning):
+    return provisioning["template"]
+
+
+@pytest.fixture(scope='module')
+def test_domain(appliance):
+    domain = appliance.collections.domains.create('test_' + fauxfactory.gen_alphanumeric(),
+                                                  'description_' + fauxfactory.gen_alphanumeric(),
+                                                  enabled=True)
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_infra_quota_via_services.py]: **REQUIRED**
Please use `format` here and further.

File: cfme/tests/infrastructure/test_infra_quota_via_services.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+@pytest.mark.meta(blockers=[BZ(1497912, forced_streams=['5.7', '5.8', '5.9', 'upstream'])])
+@pytest.mark.parametrize(
+    ['max_field_method', 'dialog_with_field', 'prov_data'],
+    [
+        [('max_vms', '1'), 'number_of_vms', {'catalog': {'vm_name': vm_name, 'num_vms': '2'}}],
+        [('max_cpus', '1'), 'number_of_sockets', {'catalog': {'vm_name': vm_name},
+                                                  'hardware': {'num_sockets': '2'}}]
+    ],
+    indirect=['max_field_method', 'dialog_with_field'],
+    ids=['max_vms', 'max_cpus'])
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_infra_quota_via_services.py]: **REQUIRED**
It looks scary, can you somehow simplify it?

Comment (vatsalparekh) [reply cfme/tests/infrastructure/test_infra_quota_via_services.py]: Can you explain what scared you?

**Comment by vatsalparekh** [comment]
Closing this as test of this PR was merged as part of other PRs

**Pull request closed without merging**